app/components/data-download.tsx

```diff
@@ -199,27 +182,40 @@ const DataDownloadMenuSection = ({
 
 const DownloadMenuItem = ({
   fileName,
-  data,
-  columnKeys,
   fileFormat,
+  dataSetIri,
+  filters,
   onDownloaded,
 }: {
   fileName: string;
-  data: Observation[];
-  columnKeys: string[];
   fileFormat: FileFormat;
+  dataSetIri: string;
+  filters?: QueryFilters;
   onDownloaded: () => void;
 }) => {
+  const locale = useLocale();
+  const urqlClient = useClient();
   const [isDownloading, setIsDownloading] = useState(false);
-  const download = useCallback(() => {
-    return fetch("/api/download", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({ columnKeys, data, fileFormat }),
-    }).then((res) =>
-      res.blob().then((blob) => saveAs(blob, `${fileName}.${fileFormat}`))
-    );
-  }, [columnKeys, data, fileFormat, fileName]);
+  const download = useCallback(
+    (preparedData: PreparedData) => {
+      const { columnKeys, data } = preparedData;
+
+      return fetch("/api/download", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          columnKeys,
+          data,
+          fileFormat,
+        }),
```

Comment:
~~ There is something I do not follow here, Why do we send back the data to the `/api/download` endpoint ? ~~

OK I checked in the code and it seems like we need to send back the data for the data to be converted into an Excel file. Ideally I guess all the data download should be done at the server side so that we do not have to do a download and then an upload and then again a download. What's your take on this ?

Since this PR is not about this per se, maybe we should keep the change for a next PR.

Reply:
I am not sure if caching would work in this case? For TEST environment I took the advantage of caching to fetch SparqlEditorUrl, which is the same query to fetch filtered data. I would need to check if this is the case, otherwise we would duplicate these queries – if it's not the case, then I agree it makes sense to also change that :)